File: utilslib/src/main/java/com/ashlikun/utils/assist/Averager.java
package com.ashlikun.utils.assist;

import com.ashlikun.utils.other.LogUtils;

import java.util.ArrayList;

/**
 * 作者　　: 李坤
 * 创建时间: 2017/8/6 0006 14:22
 * <p>
 * 方法功能：用以统计平均数
 */
public class Averager {
    private ArrayList<Number> numList = new ArrayList<Number>();

    /**
     * 添加一个数字
     */
    public synchronized void add(Number num) {
        numList.add(num);
    }

    /**
     * 清除全部
     */
    public void clear() {
        numList.clear();
    }

    /**
     * 返回参与均值计算的数字个数
     */
    public Number size() {
        return numList.size();
    }

    /**
     * 获取平均数
     */
    public Number getAverage() {
        if (numList.size() == 0) {
            return 0;
        } else {
            Float sum = 0f;
            for (int i = 0, size = numList.size(); i < size; i++) {
                sum = sum.floatValue() + numList.get(i).floatValue();
            }
            return sum / numList.size();
        }
    }

    /**
     * 打印数字列
     */
    public String print() {
        String str = "PrintList(" + size() + "): " + numList;
        LogUtils.i(str);
        return str;
    }

}

File: utilslib/src/main/java/com/ashlikun/utils/assist/Averager.kt
package com.ashlikun.utils.assist

import java.util.ArrayList
import kotlin.jvm.Synchronized
import com.ashlikun.utils.other.LogUtils

/**
 * @author　　: 李坤
 * 创建时间: 2021/12/11 15:00
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：用以统计平均数
 */

class Averager {
    private val numList = ArrayList<Number>()

    /**
     * 添加一个数字
     */
    @Synchronized
    fun add(num: Number) {
        numList.add(num)
    }

    /**
     * 清除全部
     */
    fun clear() {
        numList.clear()
    }

    /**
     * 返回参与均值计算的数字个数
     */
    fun size(): Number {
        return numList.size
    }

    /**
     * 获取平均数
     */
    val average: Number
        get() = if (numList.size == 0) {
            0
        } else {
            var sum = 0f
            var i = 0
            val size = numList.size
            while (i < size) {
                sum += numList[i].toFloat()
                i++
            }
            sum / numList.size
        }

    /**
     * 打印数字列
     */
    fun print(): String {
        val str = "PrintList(" + size() + "): " + numList
        LogUtils.i(str)
        return str
    }
}